### PR TITLE
Fix underflow panic in `InitTriInfo`

### LINKED ISSUE #14874: GenerateTangents in 0.14.1 can panic even though it is supposed to return Result
## Bevy version
 0.14.1 
 
## What went wrong

MESH:: GenerateTangents() in 0.14.1 can panic even though it is supposed to return Result 

```
 thread 'Compute Task Pool (1)' panicked at /home/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_mikktspace-0.14.0/src/generated.rs:1033:15:
attempt to subtract with overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Encountered a panic in system `bevy_clay_tiles::clay_tile_block::build_tile_block_meshes`!
```

### PR DESCRIPTION
**Pull request by kumikaya**
# Objective

- Fix #14874

## Solution

- Change the place where a panic occurs from `t < iNrTrianglesIn - 1` to `t + 1 < iNrTrianglesIn`.

## Testing

- After the fix, the following code runs successfully without any panic.

```rust
use bevy::prelude::Mesh;
use bevy_render::{
    mesh::{Indices, PrimitiveTopology},
    render_asset::RenderAssetUsages,
};

const POSITIONS: &[[f32; 3]] = &[[0.0, 1.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0]];

const NORMALS: &[[f32; 3]] = &[[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]];

const INDICES: &[u32] = &[0, 1, 2];

const UVS: &[[f32; 2]] = &[[0.0, 1.0], [0.0, 0.0], [0.0, 1.0]];

fn main() {
    let mut mesh = Mesh::new(
        PrimitiveTopology::TriangleList,
        RenderAssetUsages::default(),
    );

    mesh.insert_attribute(Mesh::ATTRIBUTE_POSITION, POSITIONS.to_vec());
    mesh.insert_attribute(Mesh::ATTRIBUTE_UV_0, UVS.to_vec());
    mesh.insert_attribute(Mesh::ATTRIBUTE_NORMAL, NORMALS.to_vec());
    mesh.insert_indices(Indices::U32(INDICES.to_vec()));
    mesh.generate_tangents().ok();

}

```

## Migration Guide

- No breaking changes introduced.

### CODE REVIEW

**Comment by janhohenheim** [comment]
Note that this will add a merge conflict to https://github.com/bevyengine/bevy/pull/9050

**Review by alice-i-cecile — APPROVED** [review]

**Review by janhohenheim — COMMENTED** [review]
I'm honestly not convinced that the code in question *should* run smoothly. That mesh seems whacky and bound to cause problems down the line. @ethereumdegen's offending mesh was similarly malformed. Do we at learn print a `warn!` in this case?

File: crates/bevy_mikktspace/src/generated.rs [hunk 1]
```diff
@@ -263,7 +263,7 @@ pub unsafe fn genTangSpace<I: Geometry>(geometry: &mut I, fAngularThreshold: f32
         pTriInfos.as_mut_ptr(),
         piTriListIn.as_ptr(),
         geometry,
-        iNrTrianglesIn,
+        iNrTrianglesIn as i32,
```
Comment (janhohenheim) [inline crates/bevy_mikktspace/src/generated.rs:266]: Isn't the data type inferred to be `i32` already? If not, what is the data type and why does the line number 255 exactly do when going below 0?

File: crates/bevy_mikktspace/src/generated.rs [hunk 2]
```diff
@@ -1077,13 +1077,8 @@ unsafe fn InitTriInfo<I: Geometry>(
...
     }
 
-    let mut pEdges = vec![SEdge::zero(); iNrTrianglesIn * 3];
-    BuildNeighborsFast(
-        pTriInfos,
-        pEdges.as_mut_ptr(),
-        piTriListIn,
-        iNrTrianglesIn as i32,
-    );
+    let mut pEdges = vec![SEdge::zero(); (iNrTrianglesIn as usize) * 3];
```
Comment (janhohenheim) [inline crates/bevy_mikktspace/src/generated.rs:1080]: I assume safety checks happened before this line?

**Comment by ethereumdegen** [comment]
> I'm honestly not convinced that the code in question *should* run smoothly. That mesh seems whacky and bound to cause problems down the line. @ethereumdegen's offending mesh was similarly malformed. Do we at least print a `warn!` in this case?


It doesnt have to run smoothly it can throw an error from the result 

But it shouldnt crash the entire program

**Comment by ethereumdegen** [comment]
Yeah my pr is about generate_tangents not mesh new

**Comment by janhohenheim** [comment]
@ethereumdegen Ooooooh I see, I didn't scroll down enough on the example code, haha. That makes sense then :)

**Comment by kumikaya** [comment]
Maybe it would be more appropriate to change the place where a panic occurs from `t < iNrTrianglesIn - 1` to `t + 1 < iNrTrianglesIn`.

**Review by janhohenheim — CHANGES_REQUESTED** [review]
@kumikaya I would prefer it if there was no panic at all, and instead `generate_tangents` returned an error in this case.
Also, no need to force push your commits to clean the history; the PR get squashed anyways and by leaving the history intact you make it easier for future readers :)

**Comment by kumikaya** [comment]
@janhohenheim Yeah, it would be better to check if `iNrTrianglesIn` is zero and return false after `iNrTrianglesIn = iTotTris - iDegenTriangles`, but I'm not quite sure at which point it would be best to return; thanks for your advice.

**Comment by janhohenheim** [comment]
@kumikaya feel free to ping me when the code you posted in the PR description results in `generate_tangents` returning an `Err` variant :)

**Comment by kumikaya** [comment]
@janhohenheim I've updated the code in the PR

**Review by janhohenheim — APPROVED** [review]
Looks good to me then :)
Thanks for fixing the crash!

**Pull request merged**